tests/CMakeLists.txt

```
@@ -0,0 +1,8 @@
+add_executable(unittests
+  tests_main.cpp
+  test_kinematics.cpp
+  )
+
+target_link_libraries(unittests EDM4HEP::edm4hep kinematics Catch2::Catch2)
```

```suggestion
target_link_libraries(unittests EDM4HEP::edm4hep kinematics Catch2::Catch2 Catch2::Catch2WithMain )
```
Requires current Catch2 devel HEAD or at least commit  4b51d0dd3be90f4faf3c10f60d1923ad47a792d7

---

Hi @vvolkl,

Is there a (planned) release date already for Catch2 v3? I briefly googled this and couldn't really find anything. I would currently prefer to use a released version of Catch2 and then make this switch once v3 is released. From your suggested changes and the preliminary release notes for v3, it seems that for our basic use cases in the unit tests, the switch should be pretty straight forward.

However, if we have many packages in the stack that use Catch2, and compile it very often when building the complete stack, then we could indeed safe a lot of useless compilation time. Do you have a feeling for how many packages are currently using Catch2?

---

Not enough ;) Right now just k4simdelphes. I think catch2 is a good candidate to use more broadly as a unit testing library. In generally i agree with your point about preferring released versions, however for the testing, which won't be part of a production release, I think this is not  such a big issue.

---

Yes, I agree that is a good point about not having to be as strict with versions for testing. I have built Catch2 as a static library and link against that now. In case we do not just want to include the amalgamated header it get's a bit trickier to include just the right headers, but I think that is something we can live with.